test-framework/junit5-component/src/main/java/io/quarkus/test/component/ConfigBeanCreator.java

```diff
@@ -0,0 +1,33 @@
+package io.quarkus.test.component;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+
+import io.quarkus.arc.BeanCreator;
+import io.quarkus.arc.SyntheticCreationalContext;
+
+public class ConfigBeanCreator implements BeanCreator<Config> {
+
+    // we need to keep a reference to the CL used to register the config object in order to support the continuous testing where TCCL does not work for us
+    private static final AtomicReference<ClassLoader> configClassLoader = new AtomicReference<>();
```

Comment:
@stuartwdouglas This workaround fixed the problem with the TCCL which is (for some reason) not the one we set  [here](https://github.com/quarkusio/quarkus/pull/33607/files#diff-1204dc0ccb675fb2a257de19a9b5104adc13bb89b7cdcde46f8b043f94d66e44R659) during continuous testing. I'm not sure if it's the best solution but it works :shrug:

Reply:
I think it is because of this line: https://github.com/quarkusio/quarkus/blob/84bcdca5badb7be24c12889b6cf7d2df1e9d9b04/core/deployment/src/main/java/io/quarkus/deployment/dev/testing/JunitTestRunner.java#L240

It looks like it was added before the TCCL was set properly before the test run: https://github.com/quarkusio/quarkus/blob/84bcdca5badb7be24c12889b6cf7d2df1e9d9b04/core/deployment/src/main/java/io/quarkus/deployment/dev/testing/JunitTestRunner.java#L223

I think we can probably remove that line.

Reply:
Thanks Stuart. I'll keep the workaroud for now. It should not hurt anything...